### DISCUSSION TITLE
a question about static file path

### DISCUSSION THREAD

**Discussion by zgangguoguo** [rwf2/Rocket#2192] [Questions] [answered]
In **main.rs**:
`.mount("/static",FileServer::from("./static"))`

page in `127.0.0.1:8000/` 
`  <script src="static/js/js.js"></script>`
it is ok

but page in `127.0.0.1:8000/article`
`  <script src="static/js/js.js"></script>`
it is 404 not found

I can add some code to address this issue  In **main.rs**:
`.mount("/article/static",FileServer::from("./static"))`

But I'd still like to know if there's a more elegant solution, similar to FLASK's **url_for** function

**Comment by SergioBenitez** [comment, marked as answer]
> In **main.rs**: `.mount("/static",FileServer::from("./static"))`
> 
> page in `127.0.0.1:8000/` ` <script src="static/js/js.js"></script>` it is ok
> 
> but page in `127.0.0.1:8000/article` ` <script src="static/js/js.js"></script>` it is 404 not found
> 
> I can add some code to address this issue In **main.rs**: `.mount("/article/static",FileServer::from("./static"))`

Just use absolute paths in your HTML: 

```diff
- <script src="static/js/js.js"></script>
+ <script src="/static/js/js.js"></script>
```

That will always resolve to the same thing and work anywhere.

> 
> But I'd still like to know if there's a more elegant solution, similar to FLASK's **url_for** function

No, but you can implement it yourself as a template helper. See the [`dyn_templates`](https://api.rocket.rs/v0.5-rc/rocket_dyn_templates/index.html#template-fairing-and-customization) docs.

**Reply by zgangguoguo** [reply]
thank you for your reply
